src/rdfTypes/Variable.cpp

```diff
@@ -61,19 +63,45 @@ Variable Variable::getWordScoreVariable(std::string_view word,
 
 // _____________________________________________________________________________
 Variable Variable::getMatchingWordVariable(std::string_view term) const {
-  return Variable{
-      absl::StrCat(MATCHINGWORD_VARIABLE_PREFIX, name().substr(1), "_", term)};
+  auto s = absl::StrCat(MATCHINGWORD_VARIABLE_PREFIX, name().substr(1), "_");
+  appendEscapedWord(term, s);
+  return Variable{std::move(s)};
 }
 
+namespace {
+// Returns true for a  subset of characters that are valid in variable names.
+// This roughly corresponds to `PN_CHARS_BASE` from the SPARQL 1.1 grammar with
+// the characters 0-9 also being allowed.
```

Comment:
Also comment that this deliberately doesn't contain `_` as we use that for escaping.